### PR TITLE
Page List: add convert panel to Inspector Controls when within Nav block

### PR DESCRIPTION
**Pull request by getdave**
## What?

Adds a new panel to the Page List block's inspector controls to allow for converting page list to navigation links. Only shown when block is within a Navigation block.

Alternative to https://github.com/WordPress/gutenberg/pull/45984/

## Why?

When the Page List is shown within the Nav Offcanvas then we need a means to convert it to links without having to go over to the editor _canvas_.

The original suggestion was to bring up the "Convert to links" modal that appears when you click `Edit` in the block toolbar within the canvas. However this had several disadvantages, namely:

- was an exception to the standard UX of clicking "edit" in the offcanvas and seeing the given block's inspector controls. This was confusing.
- would have denied access to controls within the Page List block inspector controls (for example selecting the Parent Page).
- required large amounts of coupling of the Page List block to the block-editor package.
- required lots of copy/paste of code from Page List block to Offcanvas experiment

The solution in this PR ensures a standardised user experience for all blocks within the offcanvas. Page List is no exception.

Now clicking the edit button in the offcanvas will show the inspector controls of the Page List at the top of which is the option to Customise the Menu. 

## How?

- Add panel
- Import correct utility functions
- Select parent Nav block

Downside of this PR as it stands is yet more coupling between Nav and Page List blocks.

## Testing Instructions

1. Delete all Nav Menus and Classic Menus.
2. Turn on Nav Offcanvas experiment
3. New Post -> Insert Nav block.
4. Click Nav block and open offcanvas within inspector controls.
5. Click the "Edit" icon next to the Page List block in the offcanvas.
6. See Page List block's inspector controls displayed.
7. Click on `Customise this menu`.
8. See that Page List is successfully converted _and_ the Nav block is selected.

## Screenshots or screencast 

https://user-images.githubusercontent.com/444434/206148580-443b53a6-b447-48df-b793-61e6f5bdfda5.mp4

### CODE REVIEW

**Comment by scruffian** [comment]
cc @jasmussen and @SaxonF

**Comment by scruffian** [comment]
Closed by accident

File: packages/block-library/src/page-list/edit.js [hunk 1]
```diff
@@ -173,9 +176,50 @@ export default function PageListEdit( {
...
+	}, [] );
+
 	return (
 		<>
 			<InspectorControls>
+				{ isNavigationChild && (
+					<PanelBody title={ __( 'Customize this menu' ) }>
+						<p id={ 'wp-block-page-list-modal__description' }>
+							{ __(
+								'This menu is automatically kept in sync with pages on your site. You can manage the menu yourself by clicking customize below.'
```
Comment (scruffian) [inline packages/block-library/src/page-list/edit.js:203]: Is it worth extracting this paragraph to a shared component, so that we only have one copy?

**Review by scruffian — APPROVED** [review]
LGTM

**Review by MaggieCabrera — APPROVED** [review]
I prefer this solution too. It's simpler and lets the user customize the parent of the page list.

**Pull request merged**